graphql/client.go

```diff
@@ -177,6 +191,148 @@ func (c *client) MakeRequest(ctx context.Context, req *Request, resp *Response)
 	return nil
 }
 
+type fileVariable struct {
+	mapKey string
+	file   Upload
+}
+
+// recursively find all the fields that are type Upload.
+func findFiles(parentKey string, v reflect.Value) []*fileVariable {
```

Comment:
Be aware that this could result in a stack overflow error. Please make sure that you have measures to prevent this error.